### DISCUSSION TITLE
How to interface with Arduino Mega?

### DISCUSSION THREAD

**Discussion by zhangvicto** [FYSETC/FYSETC-SPIDER#89] [Q&A]
Hi, 

I am using the Spider to run a custom 3d-printer. However, I have enclosure heating which requires many more pins than what the board offers (I plan on using 4 thermistors for the enclosure alone). I was hoping to have the separate Arduino Mega connected to the Spider and run very simple Gcode commands from the printer to use functions coded into the other board. 

I also have a Raspberry Pi but since the RPI does not come with an ADC it is hard to use the NTC100k thermistors that I am using to measure enclosure temps directly. Perhaps I could connect the Arduino Mega with IC2 to the RPI and the Spider with UART to the RPI and run everything that way? 

**What is the simplest way to do this? Am I overcomplicating things?**

Also, I was wondering what is the best method to have custom code within marlin to add functionality? - There are a lot of mixed answers on the internet. 

Thanks so much.

**Comment by ejhoness** [comment]
I think that with klipper running on Raspberry it will be possible to use subsequent controllers
